src/zjs_net.c

```diff
@@ -259,13 +259,12 @@ static void tcp_received(struct net_context *context,
 {
     sock_handle_t *handle = (sock_handle_t *)user_data;
 
-    if (handle) {
+    if (handle && buf) {
         start_socket_timeout(handle, handle->timeout);
 
         uint32_t len = net_nbuf_appdatalen(buf);
         uint8_t *data = net_nbuf_appdata(buf);
 
-        DBG_PRINT("received data, context=%p, data=%p, len=%u\n", context, data, len);
         // TODO: Probably not the right way to check if a socket is closed
```

Comment:
Yeah that was exactly what I saw too. The buf was valid but there was no data inside. And no explanation to what it meant. I think you are right that buf == NULL means the socket was closed, but I would be curious to find out what the other cases mean.

I made the assumptions for websockets that if the data pointer inside the buffer was != NULL, but length was zero, that means it was a TCP ACK. This was purely based off trial and error.

Assuming:
`buf` = argument passed into read callback
`data` = net_nbuf_appdata(buf)
`len` = net_nbuf_appdatalen(buf)

There are 4 cases I have seen:
1. `buf` == NULL
     Means socket has been closed
2. `buf` != NULL, `data` != NULL, `len` == 0
     ??? (Maybe an ACK, or some other control packet)
3. `buf` != NULL, `data` == NULL, `len` == 0
    ???
4. `buf` != NULL, `data` != NULL, `len` != 0
    Data received